firmware/applications/SignBuddy/tsc.c

```diff
@@ -92,57 +205,83 @@ static int read_value(uint32_t *value)
   }
 
   __HAL_TSC_CLEAR_FLAG(&s.tsc, (TSC_FLAG_EOA | TSC_FLAG_MCE));
-
-
-  if (HAL_TSC_GroupGetStatus(&s.tsc, TSC_GROUP3_IDX) == TSC_GROUP_COMPLETED) {
-    *value = HAL_TSC_GroupGetValue(&s.tsc, TSC_GROUP3_IDX);
-  }
   return RET_OK;
 }
 
 static inline void sample_data(void)
 {
-  uint32_t value;
+  memset(s.touch_values, 0, sizeof(s.touch_values));
 
-  if (read_value(&value) == RET_OK) {
-    //LOG_DEBUG("tsc value: %lu cal: %lu\r\n", value, s.calibration_value);
-    s.touch_value = convert_value_to_touch(value);
+  if (run_sampler() == RET_OK) {
+    get_channel_values(CHANNEL0_MIN_GROUP, CHANNEL0_MAX_GROUP, 0);
   }
   else {
-    s.touch_value = 0;
+    LOG_WARN("WARNING: TSC MAX CNT REACHED\r\n");
   }
 
-  io_config();
+  io_config(CHANNEL1);
+  HAL_TSC_IODischarge(&s.tsc, ENABLE);
+  rtos_delay_ms(1);
+
+  if (run_sampler() == RET_OK) {
+    get_channel_values(CHANNEL1_MIN_GROUP, CHANNEL1_MAX_GROUP, CHANNEL0_CNT);
+  }
+  else {
+    LOG_WARN("WARNING: TSC MAX CNT REACHED\r\n");
+  }
+
+
+  io_config(CHANNEL0);
   HAL_TSC_IODischarge(&s.tsc, ENABLE);
 }
 
+static void calibrate_channel_values(uint8_t min_group, uint8_t max_group, uint8_t offset)
+{
+  uint32_t value;
+  int index = offset;
+
+  for (int i = min_group; i <= max_group; i++) {
+    get_group_value(i, &value);
+    s.calibration_values[index] += value / CALIBRATION_SAMPLES;
+    index++;
+  }
+}
+
```

Comment:
Move this functionality into calibrate, and add these arguments to calibrate function as per the other review comment